connector/http-agent/app/Utils.js

```diff
@@ -14,9 +35,15 @@
  */
 const parseTimestamp = (ts) => {
   if (ts) {
-    const parsedTimestamp = Date.parse(ts);
-    if (Number.isNaN(parsedTimestamp)) return new Date().getTime();
-    return parsedTimestamp;
+    if (isInteger(ts)) {
+      const isValidDate = (new Date(parseInt(ts, 10))).getTime() > 0;
+      if (isValidDate) return new Date(parseInt(ts, 10)).getTime();
+    } else {
+      const isValidDate = (new Date(ts)).getTime() > 0;
+      if (isValidDate) return new Date(ts).getTime();
+    }
+    err.message = 'Invalid timestamp';
+    throw err;
```

Comment:
Vou escrever em portugues aqui para deixar melhor explicado.

Essa função está muito complicada de ler, apesar de em teoria fazer algo simples. Normalmente tente evitar ifs aninhados, além de abrir um fluxo a mais de operação, é prejudicial para entender. Isso é possivel de ser evitado por exemplo:
// if wasn't received ts value, return the current timestamp;
if (!ts){
     return new Date().getTime();
}

No caso colocar alguns comentários pode te ajudar a ver o fluxo da operação, nem sempre é bom por comentários (os principios do clean code evita), mas no começo não tem problemas. 

Além, tem código repetido como o 'new Date(parseInt(ts, 10)).getTime();', é melhor colocar ele em uma variavel auxiliar (com um bom nome) e fazer as operações com ele. Além de tudo, essa chamada cria um novo objeto e faz um parse, o que gasta recurso a toa da maquina. 


Por fim, melhor arrumar o ts antes de fazer as operações do que fazer 'if se ele é string ou inteiro'. Faz um parse dele para Inteiro antes e faz um fluxo unico depois.